test/responses_unit_test.go

```diff
@@ -0,0 +1,222 @@
+				response = nil
+				err = nil
+			} else {
+				err = fmt.Errorf("Expected error here")
+			}
+		default:
+			err = fmt.Errorf("unknown endpoint: %s", endpoint)
+		}
+	case "algod":
+		switch endpoint {
+		case "GetStatus":
+			response, err = algodC.Status().Do(context.Background())
+		case "WaitForBlock":
+			response, err = algodC.StatusAfterBlock(10).Do(context.Background())
+		case "TealCompile":
+			response, err = algodC.TealCompile([]byte{}).Do(context.Background())
+		case "RawTransaction":
+			var returnedTxid string
+			returnedTxid, err = algodC.SendRawTransaction([]byte{}).Do(context.Background())
+			response = txidresponse{TxID: returnedTxid}
+		case "GetSupply":
+			response, err = algodC.Supply().Do(context.Background())
+		case "TransactionParams":
+			var sParams types.SuggestedParams
+			sParams, err = algodC.SuggestedParams().Do(context.Background())
+			response = models.TransactionParams{
+				ConsensusVersion: sParams.ConsensusVersion,
+				Fee:              uint64(sParams.Fee),
+				GenesisID:        sParams.GenesisID,
+				Genesishash:      sParams.GenesisHash,
+				LastRound:        uint64(sParams.FirstRoundValid),
+				MinFee:           81560,
+			}
+		case "GetApplicationByID":
+			response, err = algodC.GetApplicationByID(10).Do(context.Background())
+		case "GetAssetByID":
+			response, err = algodC.GetAssetByID(10).Do(context.Background())
+		case "any":
+			response, err = indexerC.SearchForTransactions().Do(context.Background())
+			if err == nil {
+				fmt.Errorf("Expected error here")
+			}
+			if err == nil {
+				err = fmt.Errorf("Expected error here")
+			}
+			exptErr := fmt.Sprintf("HTTP %v", expectedStatus)
+			if strings.Contains(err.Error(), exptErr) {
+				response = nil
+				err = nil
+			} else {
+				err = fmt.Errorf("Expected error here")
+			}
+
+		default:
+			err = fmt.Errorf("unknown endpoint: %s", endpoint)
+		}
+	}
+	return err
+}
+
+type txidresponse struct {
+	TxID string `json:"txId"`
+}
+
+func theParsedResponseShouldEqualTheMockResponse() error {
+	var err error
+
+	if expectedStatus != 200 && response != nil {
+		return fmt.Errorf("response should be nil for response status: %d", expectedStatus)
+	}
+	if expectedStatus != 200 && response == nil {
+		return nil
+	}
+
+	responseJson := string(json.Encode(response))
+
+	jsonfile, err := os.Open(baselinePath)
+	if err != nil {
+		return err
+	}
+	fileBytes, err := ioutil.ReadAll(jsonfile)
+	if err != nil {
+		return err
+	}
+	_, err = EqualJson(string(fileBytes), responseJson)
+	return err
+}
+
+func ResponsesContext(s *godog.Suite) {
+	s.Step(`^mock http responses in "([^"]*)" loaded from "([^"]*)" with status (\d+)\.$`, mockHttpResponsesInLoadedFromWithStatus)
+	s.Step(`^we make any "([^"]*)" call to "([^"]*)"\.$`, weMakeAnyCallTo)
+	s.Step(`^the parsed response should equal the mock response\.$`, theParsedResponseShouldEqualTheMockResponse)
+}
+
+func EqualJson(j1, j2 string) (ans bool, err error) {
```

Comment:
This looks really nice. I think I understand whats happening, but could you add a couple comments to explain whats happening?

Doesn't the removed/added case need to be updated to verify that the other field is a zero value?

Reply:
I think it would be worth updating some of the old tests in order to test this function, let me know what you think.

You already updated the common `mockHttpResponsesInLoadedFrom` function, so I think it's a matter of:
```diff
func weMakeAnyPendingTransactionsInformationCall() error {    
-  algodClient, err := algod.MakeClient(mockServer.URL, "")    
-  if err != nil {    
-    return err    
-  }      
-  _, stxsResponse, globalErrForExamination = algodClient.PendingTransactions().Do(context.Background())    
-  return nil    
+  weMakeAnyCallTo("algod", "PendingTransactions")
+  return nil
}    

func theParsedPendingTransactionsInformationResponseShouldContainAnArrayOfLenAndElementNumberShouldHaveSender(length, idx int, sender string) error {
-  if len(stxsResponse) != length {
-    return fmt.Errorf("expected response length %d but received length %d", length, len(stxsResponse))
-  }
-  if stxsResponse[idx].Txn.Sender.String() != sender {
-    return fmt.Errorf("expected txn %d to have sender %s but real sender was %s", idx, sender, stxsResponse[idx].Txn.Sender.String())
-  }
+  theParsedResponseShouldEqualTheMockResponse()
  return nil
}
```

This might be a bad example, I think `PendingTransactions` is one of the `Format=msgpack` endpoints.

Reply:
Let's leave the code cleanup for later, since we are on tight schedule right now. 
I'll add the comments :-)

Reply:
> Doesn't the removed/added case need to be updated to verify that the other field is a zero value?

We will add it as we get a case for it. Right now, non is observed.